### PR TITLE
Switch payment method when adding

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Switch payment method when adding

Use add payment method response to get id

[stage-2]

## Motivation and Context
Billing frictions epic.

## How Has This Been Tested?
Tested changes locally. Updated unit tests.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

File: web/scripts/billing/services/svc-add-payment-source-factory.js [hunk 1]
```diff
@@ -58,7 +58,15 @@ angular.module('risevision.apps.billing.services')
...
-        return billing.addPaymentSource(intentId);
+        return billing.addPaymentSource(intentId)
+          .then(function (response) {
+            return response && response.item && response.item.payment_source &&
+              response.item.payment_source.id;
+          });
+      };
+
+      var _changePaymentSource = function (subscriptionId, paymentSourceId) {
+        return billing.changePaymentSource(subscriptionId, paymentSourceId);
```
Comment (alex-deaconu) [inline web/scripts/billing/services/svc-add-payment-source-factory.js:69]: Calling service directly. `subscriptionFactory` has the same method we don't need the overhead here.

File: web/scripts/billing/services/svc-add-payment-source-factory.js [hunk 2]
```diff
@@ -58,7 +58,15 @@ angular.module('risevision.apps.billing.services')
         var intentId = creditCardFactory.paymentMethods.intentResponse ? 
           creditCardFactory.paymentMethods.intentResponse.intentId : null;
 
-        return billing.addPaymentSource(intentId);
+        return billing.addPaymentSource(intentId)
+          .then(function (response) {
+            return response && response.item && response.item.payment_source &&
+              response.item.payment_source.id;
```
Comment (alex-deaconu) [inline web/scripts/billing/services/svc-add-payment-source-factory.js:64]: For some reason, the Stripe `paymentMethodResponse`'s `id` is different and does not work. We need to use the `id` returned from Chargebee.

**Comment by alex-deaconu** [comment]
@ezequielc please review. Thanks!

File: web/scripts/billing/services/svc-add-payment-source-factory.js (hunk 2 above)
Comment (ezequielc) [reply web/scripts/billing/services/svc-add-payment-source-factory.js:64]: Yeah, they are different. Chargebee creates is own id. Thanks!

**Review by ezequielc — APPROVED** [review]

**Pull request merged**